core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -155,6 +158,9 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           // in this block are read when requesting executors
           CoarseGrainedSchedulerBackend.this.synchronized {
             executorDataMap.put(executorId, data)
+            if (currentExecutorIdCounter < Integer.parseInt(executorId)) {
+              currentExecutorIdCounter = Integer.parseInt(executorId)
+            }
```

Comment:
yes, with dynamic allocation, for example, the executor with the max known id may be gone already.

minor: `executorId.toInt` instead of `Integer.parseInt`